### PR TITLE
Make area chart keyboard accessible

### PR DESCRIPTION
**Pull request by krkshitij**
## Previous Behavior

Data points in AreaChart are not accessible using keyboard

## New Behavior

Data points in AreaChart are accessible using keyboard

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/5ac5a3d5e401ca41ba30d3facfda9104357ae34a'>5ac5a3d5e401ca41ba30d3facfda9104357ae34a</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 1]
```diff
@@ -598,46 +600,90 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
...
+        );
+      });
+    } else {
+      // Render data points close to the mouse pointer for large dataset
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      this._stackedData.forEach((singleStackedData: Array<any>, index: number) => {
+        if (points.length === index) {
+          return;
+        }
+        singleStackedData.forEach((singlePoint: IDPointType, pointIndex: number) => {
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:662]: why did you remove <g> for large datapoints

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 2]
```diff
@@ -598,46 +600,90 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
...
             );
-          })}
-        </g>,
-      );
-    });
+          }
+        });
+      });
+    }
     graph.push(
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:687]: do both approaches still work when there is only 1 point in the chart

File: packages/react-examples/src/react-charting/AreaChart/AreaChart.Multiple.Example.tsx [hunk 3]
```diff
@@ -211,6 +211,7 @@ export class AreaChartMultipleExample extends React.Component<{}, IAreaChartBasi
             legendProps={{
               allowFocusOnLegends: true,
             }}
+            optimizeLargeData
```
Comment (AtishayMsft) [inline packages/react-examples/src/react-charting/AreaChart/AreaChart.Multiple.Example.tsx:214]: Add a label that this chart is using optimizeLargeData flag.

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 4]
```diff
@@ -598,46 +600,90 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
...
+                  data-is-focusable={true}
+                  cx={xScale(singlePoint.xVal)}
+                  cy={yScale(singlePoint.values[1])}
+                  stroke={lineColor}
+                  strokeWidth={3}
+                  fill={this._updateCircleFillColor(xDataPoint, lineColor, circleId)}
+                  onMouseOut={this._onRectMouseOut}
+                  onMouseOver={this._onRectMouseMove}
+                  onClick={this._onDataPointClick.bind(this, points[index]!.data[pointIndex].onDataPointClick!)}
+                  onFocus={() => this._handleFocus(index, pointIndex, circleId)}
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:643]: I am seeing 1 issue.
Click on a datapoint. Then hover on previous points. The hover card continues to render for the datapoint that was clicked earlier.

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx (hunk 1 above)
Comment (AtishayMsft) [reply packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:662]: Figured it out.

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx [hunk 5]
```diff
@@ -598,46 +600,90 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
...
+                  role="img"
+                  aria-label={this._getAriaLabel(index, pointIndex)}
+                />
+              );
+            })}
+          </g>,
+        );
+      });
+    } else {
+      // Render data points close to the mouse pointer for large dataset
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/AreaChart/AreaChart.base.tsx:656]: This comment is not clear. We are only changing how circles are rendered not data points.

File: packages/react-examples/src/react-charting/AreaChart/AreaChart.Multiple.Example.tsx (hunk 3 above)
Comment (krkshitij) [reply packages/react-examples/src/react-charting/AreaChart/AreaChart.Multiple.Example.tsx]: added separate example for it

File: packages/react-charting/src/components/AreaChart/AreaChart.base.tsx (hunk 5 above)
Comment (krkshitij) [reply packages/react-charting/src/components/AreaChart/AreaChart.base.tsx]: fixed

File: packages/react-examples/src/react-charting/AreaChart/AreaChart.LargeData.Example.tsx
Comment (Hotell) [inline packages/react-examples/src/react-charting/AreaChart/AreaChart.LargeData.Example.tsx:1]: looks like these 2 files have no owner (which is a red flag in terms of maintenance and security).

image


Can pls you setup one in follow up PR ?

**Review by Hotell — APPROVED** [review]
